src/commands/cmd_query.c

```diff
@@ -31,30 +23,24 @@ static void _index_operation(RedisModuleCtx *ctx, GraphContext *gc,
 														  index_op));
 		const char *prop = cypher_ast_prop_name_get_value(cypher_ast_create_node_props_index_get_prop_name(
 															  index_op, 0));
-		if(GraphContext_AddIndex(&idx, gc, label, prop, IDX_EXACT_MATCH) != INDEX_OK) {
-			// Index creation may have failed if the label or property was invalid, or the index already exists.
-			RedisModule_ReplyWithSimpleString(ctx, "(no changes, no records)");
-		} else {
-			Index_Construct(idx);
-			RedisModule_ReplyWithSimpleString(ctx, "Indices added: 1");
-		}
+		QueryCtx_LockForCommit();
+		if(GraphContext_AddIndex(&idx, gc, label, prop, IDX_EXACT_MATCH) == INDEX_OK) Index_Construct(idx);
+		QueryCtx_UnlockCommit(NULL);
 	} else {
 		// Retrieve strings from AST node
 		const char *label = cypher_ast_label_get_name(cypher_ast_drop_node_props_index_get_label(index_op));
 		const char *prop = cypher_ast_prop_name_get_value(cypher_ast_drop_node_props_index_get_prop_name(
 															  index_op, 0));
-		if(GraphContext_DeleteIndex(gc, label, prop, IDX_EXACT_MATCH) == INDEX_OK) {
-			RedisModule_ReplyWithSimpleString(ctx, "Indices removed: 1");
-		} else {
-			char *reply;
-			asprintf(&reply, "ERR Unable to drop index on :%s(%s): no such index.", label, prop);
-			RedisModule_ReplyWithError(ctx, reply);
-			free(reply);
+		QueryCtx_LockForCommit();
+		int res = GraphContext_DeleteIndex(gc, label, prop, IDX_EXACT_MATCH);
+		QueryCtx_UnlockCommit(NULL);
+
+		if(res != INDEX_OK) {
+			char *error;
+			asprintf(&error, "ERR Unable to drop index on :%s(%s): no such index.", label, prop);
+			QueryCtx_SetError(error);
 		}
```

Comment:
Why do we check `res` and set an error in the case of index delete and not in index create?